### PR TITLE
move tests to testament

### PR DESCRIPTION
**Pull request by ringabout**
- change two or three unittest style to doAssert
- use include because some symbols are not exported
- disable testing random module

### CODE REVIEW

File: lib/pure/algorithm.nim [hunk 1]
```diff
@@ -665,36 +665,6 @@ proc prevPermutation*[T](x: var openArray[T]): bool {.discardable.} =
 
   result = true
 
-when isMainModule:
```
Comment (timotheecour) [inline lib/pure/algorithm.nim:668]: @xflywind 
after PR we now have both tests/stdlib/ttables.nim and tests/collections/ttables.nim
(but it was indeed better to delay fixing this to a separate PR so this one stays a simple move operation)

these should be merged into a single one

**Comment by timotheecour** [comment]
to review this PR, you really need `git diff --color-moved=blocks --color-moved-ws=ignore-all-space` as recommended here https://github.com/nim-lang/Nim/pull/13162 otherwise github / diff without options won't tell you what actually changed vs what was moved...

File: lib/pure/unidecode/unidecode.nim [hunk 2]
```diff
@@ -66,8 +66,3 @@ proc unidecode*(s: string): string =
     var c = int(r)
     if c <=% 127: add(result, chr(c))
     elif c <% translationTable.len: add(result, translationTable[c-128])
-
-when isMainModule:
-  #loadUnidecodeTable("lib/pure/unidecode/unidecode.dat")
-  doAssert unidecode("Äußerst") == "Ausserst"
-  doAssert unidecode("北京") == "Bei Jing "
```
Comment (timotheecour) [inline lib/pure/unidecode/unidecode.nim:73]: why were those removed?

File: tests/stdlib/turi.nim [hunk 3]
```diff
@@ -15,3 +15,286 @@ block:
...
+  # bug #3207
+  block:
+    doAssert parseUri("http://qq/1").combine(parseUri("https://qqq")).`$` == "https://qqq"
+
+  # bug #4959
+  block:
+    let foo = parseUri("http://example.com") / "/baz"
+    doAssert foo.path == "/baz"
+
+  # bug found on stream 13/10/17
```
Comment (timotheecour) [inline tests/stdlib/turi.nim:194]: for next time, prefer this style in 1 line:
```nim
block: # bug found on stream 13/10/17
```
* more consistent with block logic
* more similar to `block: label`
* more similar to `test "VM and runtime should` in unittests
* more common and shorter

File: tests/stdlib/thttpcore.nim [hunk 4]
```diff
@@ -52,7 +45,7 @@ suite "httpcore":
     doAssert parseHeader("Accept: foo, bar, prologue") == (key: "Accept", value: @["foo", "bar", "prologue"])
     doAssert parseHeader("Accept: foo, bar, prologue, starlight") == (key: "Accept", value: @["foo", "bar", "prologue", "starlight"])
 
-  test "add empty sequence to HTTP headers":
+  block add_empty_sequence_to_HTTP_headers:
```
Comment (timotheecour) [inline tests/stdlib/thttpcore.nim:48]: prefer this style:
```nim
block: "add empty sequence to HTTP headers"
```

File: lib/pure/unidecode/unidecode.nim (hunk 2 above)
Comment (ringabout) [reply lib/pure/unidecode/unidecode.nim:73]: tests already exist
https://github.com/nim-lang/Nim/blob/devel/tests/stdlib/tunidecode.nim

Comment (timotheecour) [reply lib/pure/unidecode/unidecode.nim:73]: i saw it but the ones you removed are missing; yu could simply add back these:
```nim
doAssert unidecode("Äußerst") == "Ausserst"
doAssert unidecode("北京") == "Bei Jing " # this one was removed
```
(and use doAssert instead of echo)

File: tests/stdlib/thttpcore.nim (hunk 4 above)
Comment (narimiran) [reply tests/stdlib/thttpcore.nim:48]: Why?

File: lib/pure/unidecode/unidecode.nim (hunk 2 above)
Comment (ringabout) [reply lib/pure/unidecode/unidecode.nim:73]: done https://github.com/nim-lang/Nim/pull/16157
